[Automation] javac compile fails for org.jetbrains.exposed:exposed-core:1.2.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25672076395/job/75360059967
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
: T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

> Task :compileTestKotlin FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 17s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jetbrains.exposed:exposed-core:1.0.0-beta-4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 33s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.2.0 at 2026-05-11T13:11:40Z with exit code 1 after 34s
❌ ERROR: 'native-image run' for 1.2.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean test -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
========================================================================================
 org.jetbrains.exposed:exposed-core:1.0.0-beta-4:1.2.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.2.0 at 2026-05-11T13:11:40Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 926ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

> Task :compileTestKotlin FAILED
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:19:8 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:146:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:28 'fun <T : Comparable<T>, S : T?> Expression<in S>.less(other: Expression<in S>): LessOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:77 'fun <T : String?> Expression<T>.like(pattern: String): LikeEscapeOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:65 'fun <T> ExpressionWithColumnType<T>.inList(list: Iterable<T>): InListOrNotInListBaseOp<T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:71 'fun <T> ExpressionWithColumnType<T>.plus(t: T): PlusOp<T, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:76 'fun <T, S : T> ExpressionWithColumnType<T>.times(other: Expression<S>): TimesOp<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:93 'fun <T : Number?, S : T> ExpressionWithColumnType<T>.rem(t: S): ModOp<T, S, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:157:35 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:158:13 'fun case(): Case' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:159:37 'fun <T> ExpressionWithColumnType<T>.eq(t: T): Op<Boolean>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:164:30 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 2 executed, 9 up-to-date


BUILD FAILED in 2s

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.2.0 at 2026-05-11T13:11:44Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 1.2.0 failed with exit code 1.
FAILED[javac compile][1.2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25672076395/job/75360060063
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
5:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 14s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jetbrains.exposed:exposed-core:1.0.0-beta-4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 30s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.2.0 at 2026-05-11T13:11:27Z with exit code 1 after 30s
❌ ERROR: 'native-image run' for 1.2.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean test -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
========================================================================================
 org.jetbrains.exposed:exposed-core:1.0.0-beta-4:1.2.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.2.0 at 2026-05-11T13:11:27Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 965ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

> Task :compileTestKotlin FAILED
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:19:8 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:146:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:28 'fun <T : Comparable<T>, S : T?> Expression<in S>.less(other: Expression<in S>): LessOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:77 'fun <T : String?> Expression<T>.like(pattern: String): LikeEscapeOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:65 'fun <T> ExpressionWithColumnType<T>.inList(list: Iterable<T>): InListOrNotInListBaseOp<T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:71 'fun <T> ExpressionWithColumnType<T>.plus(t: T): PlusOp<T, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:76 'fun <T, S : T> ExpressionWithColumnType<T>.times(other: Expression<S>): TimesOp<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:93 'fun <T : Number?, S : T> ExpressionWithColumnType<T>.rem(t: S): ModOp<T, S, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:157:35 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:158:13 'fun case(): Case' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:159:37 'fun <T> ExpressionWithColumnType<T>.eq(t: T): Op<Boolean>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:164:30 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 2 executed, 9 up-to-date
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.2.0 at 2026-05-11T13:11:32Z with exit code 1 after 5s
❌ ERROR: 'javac compile' for 1.2.0 failed with exit code 1.
FAILED[javac compile][1.2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25672076395/job/75360060021
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 19s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jetbrains.exposed:exposed-core:1.0.0-beta-4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 36s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.2.0 at 2026-05-11T13:11:42Z with exit code 1 after 36s
❌ ERROR: 'native-image run' for 1.2.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.2.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.2.0" ./gradlew clean test -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
========================================================================================
 org.jetbrains.exposed:exposed-core:1.0.0-beta-4:1.2.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.2.0 at 2026-05-11T13:11:42Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 910ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:19:8 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.> Task :compileTestKotlin FAILED

e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:146:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:28 'fun <T : Comparable<T>, S : T?> Expression<in S>.less(other: Expression<in S>): LessOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:77 'fun <T : String?> Expression<T>.like(pattern: String): LikeEscapeOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:65 'fun <T> ExpressionWithColumnType<T>.inList(list: Iterable<T>): InListOrNotInListBaseOp<T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:71 'fun <T> ExpressionWithColumnType<T>.plus(t: T): PlusOp<T, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:76 'fun <T, S : T> ExpressionWithColumnType<T>.times(other: Expression<S>): TimesOp<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:93 'fun <T : Number?, S : T> ExpressionWithColumnType<T>.rem(t: S): ModOp<T, S, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:157:35 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:158:13 'fun case(): Case' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:159:37 'fun <T> ExpressionWithColumnType<T>.eq(t: T): Op<Boolean>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:164:30 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 2 executed, 9 up-to-date
.

BUILD FAILED in 2s

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.2.0 at 2026-05-11T13:11:47Z with exit code 1 after 5s
❌ ERROR: 'javac compile' for 1.2.0 failed with exit code 1.
FAILED[javac compile][1.2.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.2.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
```

--- a/metadata/org.jetbrains.exposed/exposed-core/1.2.0/reachability-metadata.json
+++ b/metadata/org.jetbrains.exposed/exposed-core/1.2.0/reachability-metadata.json
@@ -1,0 +1,34 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.IntegerColumnType",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.ColumnType",
+      "methods": [
+        {
+          "name": "getNullable",
+          "parameterTypes": []
+        },
+        {
+          "name": "setNullable",
+          "parameterTypes": [
+            "boolean"
+          ]
+        }
+      ]
+    }
+  ]
+}

--- a/metadata/org.jetbrains.exposed/exposed-core/index.json
+++ b/metadata/org.jetbrains.exposed/exposed-core/index.json
@@ -20,6 +20,11 @@
   },
   {
     "metadata-version" : "1.2.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/jetbrains/exposed/exposed-core/$version$/exposed-core-$version$-sources.jar",
+    "repository-url" : "https://github.com/JetBrains/Exposed",
+    "test-code-url" : "https://github.com/JetBrains/Exposed/tree/$version$/exposed-tests/src/test",
+    "documentation-url" : "https://github.com/JetBrains/Exposed/blob/$version$/docs/api/exposed-core/index.html",
+    "description" : "Exposed Core is the base module of JetBrains Exposed, a Kotlin SQL/ORM framework for building type-safe database code. It provides the SQL DSL primitives, schema and table model, expressions, statements, and dialect abstractions used by Exposed's JDBC, R2DBC, and DAO modules.",
     "language" : {
       "name" : "kotlin",
       "version" : "2.0"

--- a/metadata/org.jetbrains.exposed/exposed-core/index.json
+++ b/metadata/org.jetbrains.exposed/exposed-core/index.json
@@ -19,6 +19,19 @@
     ]
   },
   {
+    "metadata-version" : "1.2.0",
+    "language" : {
+      "name" : "kotlin",
+      "version" : "2.0"
+    },
+    "tested-versions" : [
+      "1.2.0"
+    ],
+    "allowed-packages" : [
+      "org.jetbrains.exposed.v1"
+    ]
+  },
+  {
     "metadata-version" : "1.1.1",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/jetbrains/exposed/exposed-core/$version$/exposed-core-$version$-sources.jar",
     "repository-url" : "https://github.com/JetBrains/Exposed",

--- a/stats/org.jetbrains.exposed/exposed-core/1.2.0/execution-metrics.json
+++ b/stats/org.jetbrains.exposed/exposed-core/1.2.0/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_javac_fail:2026-05-11": {
+    "timestamp": "2026-05-11T18:52:44.542897Z",
+    "library": "org.jetbrains.exposed:exposed-core:1.2.0",
+    "starting_commit": "2d03cb03b62fa77dd5610300aa973fe29ca92076",
+    "ending_commit": "dcb48a870e80d31b65d2d86cc93c1fd5ede7db6c",
+    "previous_library": "org.jetbrains.exposed:exposed-core:1.0.0-beta-4",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "1.2.0",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4355,
+          "missed": 54890,
+          "ratio": 0.073508,
+          "total": 59245
+        },
+        "line": {
+          "covered": 703,
+          "missed": 7378,
+          "ratio": 0.086994,
+          "total": 8081
+        },
+        "method": {
+          "covered": 324,
+          "missed": 3117,
+          "ratio": 0.094159,
+          "total": 3441
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.0.0-beta-4",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4466,
+          "missed": 50240,
+          "ratio": 0.081636,
+          "total": 54706
+        },
+        "line": {
+          "covered": 710,
+          "missed": 6615,
+          "ratio": 0.096928,
+          "total": 7325
+        },
+        "method": {
+          "covered": 339,
+          "missed": 2727,
+          "ratio": 0.110568,
+          "total": 3066
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 45463,
+      "cached_input_tokens_used": 107008,
+      "output_tokens_used": 2519,
+      "iterations": 1,
+      "cost_usd": 0.1291,
+      "tested_library_loc": 703,
+      "metadata_entries": 5,
+      "previous_library_metadata_entries": 5,
+      "code_coverage_percent": 8.7,
+      "previous_library_coverage_percent": 9.69
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.jetbrains.exposed/exposed-core/1.2.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt",
+      "metadata_file": "metadata/org.jetbrains.exposed/exposed-core/1.2.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.jetbrains.exposed/exposed-core/1.2.0/stats.json
+++ b/stats/org.jetbrains.exposed/exposed-core/1.2.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "1.2.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 4355,
+        "missed" : 54890,
+        "ratio" : 0.073508,
+        "total" : 59245
+      },
+      "line" : {
+        "covered" : 703,
+        "missed" : 7378,
+        "ratio" : 0.086994,
+        "total" : 8081
+      },
+      "method" : {
+        "covered" : 324,
+        "missed" : 3117,
+        "ratio" : 0.094159,
+        "total" : 3441
+      }
+    }
+  } ]
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/.gitignore
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/build.gradle
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/build.gradle
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+    id "org.jetbrains.kotlin.jvm" version "2.3.20"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.jetbrains.exposed:exposed-core:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+kotlin {
+    jvmToolchain(21)
+}
+
+compileTestKotlin {
+    kotlinOptions.jvmTarget = "21"
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/gradle.properties
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.jetbrains.exposed:exposed-core:1.2.0
+library.version = 1.2.0
+metadata.dir = org.jetbrains.exposed/exposed-core/1.2.0/

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/settings.gradle
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.jetbrains.exposed.exposed-core_tests'

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
@@ -16,17 +16,26 @@ import org.jetbrains.exposed.v1.core.Index
 import org.jetbrains.exposed.v1.core.Join
 import org.jetbrains.exposed.v1.core.QueryBuilder
 import org.jetbrains.exposed.v1.core.Schema
-import org.jetbrains.exposed.v1.core.SqlExpressionBuilder
 import org.jetbrains.exposed.v1.core.Table
 import org.jetbrains.exposed.v1.core.Version
 import org.jetbrains.exposed.v1.core.WindowFrameBound
 import org.jetbrains.exposed.v1.core.alias
 import org.jetbrains.exposed.v1.core.and
 import org.jetbrains.exposed.v1.core.append
+import org.jetbrains.exposed.v1.core.case
+import org.jetbrains.exposed.v1.core.coalesce
+import org.jetbrains.exposed.v1.core.eq
+import org.jetbrains.exposed.v1.core.inList
 import org.jetbrains.exposed.v1.core.intLiteral
 import org.jetbrains.exposed.v1.core.intParam
+import org.jetbrains.exposed.v1.core.less
+import org.jetbrains.exposed.v1.core.like
+import org.jetbrains.exposed.v1.core.plus
+import org.jetbrains.exposed.v1.core.rem
+import org.jetbrains.exposed.v1.core.rowNumber
 import org.jetbrains.exposed.v1.core.stringLiteral
 import org.jetbrains.exposed.v1.core.stringParam
+import org.jetbrains.exposed.v1.core.times
 import org.jetbrains.exposed.v1.exceptions.DuplicateColumnException
 import org.junit.jupiter.api.Test
 
@@ -143,39 +152,31 @@ public class ExposedCoreTest {
 
     @Test
     fun expressionDslBuildsComparisonListsAliasesCasesAndWindows(): Unit {
-        val comparison = with(SqlExpressionBuilder) {
-            (intLiteral(1) less intLiteral(3)) and (stringLiteral("kotlin") like "kot%")
-        }
+        val comparison = (intLiteral(1) less intLiteral(3)) and (stringLiteral("kotlin") like "kot%")
         assertThat(comparison.toString()).isEqualTo("(1 < 3) AND ('kotlin' LIKE 'kot%')")
 
-        val inList = with(SqlExpressionBuilder) { intLiteral(2) inList listOf(3, 1, 2) }
+        val inList = intLiteral(2) inList listOf(3, 1, 2)
         assertThat(inList.toString()).isEqualTo("2 IN (3, 1, 2)")
 
-        val arithmetic = with(SqlExpressionBuilder) { ((intLiteral(5) + 4) * intLiteral(2)) % 3 }
+        val arithmetic = ((intLiteral(5) + 4) * intLiteral(2)) % 3
         assertThat(arithmetic.toString()).isEqualTo("(((5 + 4) * 2) % 3)")
 
-        val caseExpression = with(SqlExpressionBuilder) {
-            case()
-                .When(intLiteral(1) eq 1, stringLiteral("one"))
-                .Else(stringLiteral("other"))
-        }
+        val caseExpression = case()
+            .When(intLiteral(1) eq 1, stringLiteral("one"))
+            .Else(stringLiteral("other"))
         assertThat(caseExpression.toString()).isEqualTo("CASE WHEN 1 = 1 THEN 'one' ELSE 'other' END")
 
-        val coalesced = with(SqlExpressionBuilder) {
-            coalesce(stringLiteral("primary"), stringLiteral("fallback"), stringLiteral("last"))
-        }
+        val coalesced = coalesce(stringLiteral("primary"), stringLiteral("fallback"), stringLiteral("last"))
         assertThat(coalesced.toString()).isEqualTo("COALESCE('primary', 'fallback', 'last')")
 
         val aliased = intLiteral(10).alias("ten")
         assertThat(aliased.toString()).isEqualTo("10 ten")
         assertThat(aliased.aliasOnlyExpression().toString()).isEqualTo("ten")
 
-        val window = with(SqlExpressionBuilder) {
-            rowNumber()
-                .over()
-                .partitionBy(stringLiteral("partition"))
-                .rows(WindowFrameBound.unboundedPreceding())
-        }
+        val window = rowNumber()
+            .over()
+            .partitionBy(stringLiteral("partition"))
+            .rows(WindowFrameBound.unboundedPreceding())
         assertThat(window.toString()).isEqualTo("ROW_NUMBER() OVER(PARTITION BY 'partition' ROWS UNBOUNDED PRECEDING)")
     }
 

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
@@ -1,0 +1,275 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jetbrains_exposed.exposed_core
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.jetbrains.exposed.v1.core.Alias
+import org.jetbrains.exposed.v1.core.Column
+import org.jetbrains.exposed.v1.core.DatabaseConfig
+import org.jetbrains.exposed.v1.core.ExperimentalKeywordApi
+import org.jetbrains.exposed.v1.core.Index
+import org.jetbrains.exposed.v1.core.Join
+import org.jetbrains.exposed.v1.core.QueryBuilder
+import org.jetbrains.exposed.v1.core.Schema
+import org.jetbrains.exposed.v1.core.SqlExpressionBuilder
+import org.jetbrains.exposed.v1.core.Table
+import org.jetbrains.exposed.v1.core.Version
+import org.jetbrains.exposed.v1.core.WindowFrameBound
+import org.jetbrains.exposed.v1.core.alias
+import org.jetbrains.exposed.v1.core.and
+import org.jetbrains.exposed.v1.core.append
+import org.jetbrains.exposed.v1.core.intLiteral
+import org.jetbrains.exposed.v1.core.intParam
+import org.jetbrains.exposed.v1.core.stringLiteral
+import org.jetbrains.exposed.v1.core.stringParam
+import org.jetbrains.exposed.v1.exceptions.DuplicateColumnException
+import org.junit.jupiter.api.Test
+
+public class ExposedCoreTest {
+    @Test
+    fun tableMetadataTracksColumnsKeysReferencesAndAliases(): Unit {
+        assertThat(Users.tableName).isEqualTo("app.users")
+        assertThat(Users.schemaName).isEqualTo("app")
+        assertThat(Users.columns.map { it.name })
+            .containsExactly("id", "email", "display_name", "status", "score", "active", "generated_token")
+        val primaryKeyColumns: List<Column<*>> = requireNotNull(Users.primaryKey).columns.toList()
+        assertThat(primaryKeyColumns).containsExactly(Users.id)
+        assertThat(Users.autoIncColumn).isEqualTo(Users.id)
+
+        val emailIndex: Index = Users.indices.single()
+        assertThat(emailIndex.customName).isEqualTo("ux_users_email")
+        assertThat(emailIndex.unique).isTrue()
+        assertThat(emailIndex.columns).containsExactly(Users.email)
+
+        assertThat(Memberships.user.referee).isEqualTo(Users.id)
+        assertThat(Memberships.group.referee).isEqualTo(Groups.id)
+        assertThat(Memberships.foreignKeys).hasSize(2)
+
+        val userAlias: Alias<Users> = Users.alias("u")
+        val aliasedEmail: Column<String> = userAlias[Users.email]
+        assertThat(userAlias.tableName).isEqualTo("u")
+        assertThat(userAlias.tableNameWithAlias).isEqualTo("app.users u")
+        assertThat(aliasedEmail.name).isEqualTo(Users.email.name)
+        assertThat(aliasedEmail.table).isEqualTo(userAlias)
+        assertThat(userAlias.originalColumn(aliasedEmail)).isEqualTo(Users.email)
+    }
+
+    @Test
+    fun joinsInferForeignKeysAndExposeCombinedFieldSets(): Unit {
+        val userMembershipJoin: Join = Users.innerJoin(Memberships)
+
+        assertThat(userMembershipJoin.columns).containsExactlyElementsOf(Users.columns + Memberships.columns)
+        assertThat(userMembershipJoin.fields).containsExactlyElementsOf(Users.fields + Memberships.fields)
+        assertThat(userMembershipJoin.alreadyInJoin(Memberships)).isTrue()
+        assertThat(userMembershipJoin.alreadyInJoin(Groups)).isFalse()
+
+        val userMembershipGroupJoin: Join = userMembershipJoin.innerJoin(Groups)
+
+        assertThat(userMembershipGroupJoin.columns)
+            .containsExactlyElementsOf(Users.columns + Memberships.columns + Groups.columns)
+        assertThat(userMembershipGroupJoin.fields)
+            .containsExactlyElementsOf(Users.fields + Memberships.fields + Groups.fields)
+        assertThat(userMembershipGroupJoin.alreadyInJoin(Memberships)).isTrue()
+        assertThat(userMembershipGroupJoin.alreadyInJoin(Groups)).isTrue()
+    }
+
+    @Test
+    fun duplicateColumnNamesAreRejected(): Unit {
+        val table: Table = Table("duplicate_columns")
+        table.integer("id")
+
+        assertThatThrownBy { table.varchar("id", 16) }
+            .isInstanceOf(DuplicateColumnException::class.java)
+            .hasMessageContaining("id")
+            .hasMessageContaining("duplicate_columns")
+    }
+
+    @Test
+    fun columnDefaultsNullabilityAndGeneratedFlagsAreTracked(): Unit {
+        assertThat(Users.email.defaultValueFun == null).isTrue()
+        assertThat(Users.email.defaultValueInDb() == null).isTrue()
+        assertThat(Users.email.columnType.nullable).isFalse()
+
+        assertThat(Users.displayName.columnType.nullable).isTrue()
+        assertThat(Users.status.defaultValueFun?.invoke()).isEqualTo("active")
+        assertThat(Users.status.defaultValueInDb().toString()).isEqualTo("'active'")
+        assertThat(Users.score.defaultValueFun?.invoke()).isEqualTo(7)
+        assertThat(Users.score.defaultValueInDb() == null).isTrue()
+        assertThat(Users.active.defaultValueFun?.invoke()).isTrue()
+        assertThat(Users.active.defaultValueInDb()).isNotNull()
+
+        Users.generatedToken.defaultValueFun?.invoke().let { value: String? ->
+            assertThat(value).isEqualTo("generated")
+        }
+        assertThat(Users.generatedToken.isDatabaseGenerated()).isTrue()
+    }
+
+    @Test
+    fun columnTypesValidateAndFormatValues(): Unit {
+        assertThat(Users.id.columnType.valueFromDB("42")).isEqualTo(42)
+        assertThat(Users.email.columnType.valueFromDB(byteArrayOf(65, 66, 67))).isEqualTo("ABC")
+        assertThat(Users.email.columnType.valueToString("O'Reilly\nGuide")).isEqualTo("'O''Reilly\\nGuide'")
+        assertThat(Users.displayName.columnType.valueToString(null)).isEqualTo("NULL")
+
+        Users.email.columnType.validateValueBeforeUpdate("short@example.org")
+        assertThatThrownBy { Users.email.columnType.validateValueBeforeUpdate("x".repeat(129)) }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("exceeds length")
+
+        assertThatThrownBy { Users.email.columnType.valueToString(null) }
+            .isInstanceOf(IllegalStateException::class.java)
+            .hasMessageContaining("NULL in non-nullable column")
+    }
+
+    @Test
+    fun queryBuilderFormatsInlineSqlAndPreparedArguments(): Unit {
+        val inlineBuilder: QueryBuilder = QueryBuilder(prepared = false)
+        inlineBuilder.append(intLiteral(42), ", ", stringLiteral("O'Reilly"))
+
+        assertThat(inlineBuilder.toString()).isEqualTo("42, 'O''Reilly'")
+        assertThat(inlineBuilder.args).isEmpty()
+
+        val preparedBuilder: QueryBuilder = QueryBuilder(prepared = true)
+        preparedBuilder.append(intParam(7), ", ", stringParam("alpha"))
+
+        assertThat(preparedBuilder.toString()).isEqualTo("?, ?")
+        assertThat(preparedBuilder.args.map { it.second }).containsExactly(7, "alpha")
+    }
+
+    @Test
+    fun expressionDslBuildsComparisonListsAliasesCasesAndWindows(): Unit {
+        val comparison = with(SqlExpressionBuilder) {
+            (intLiteral(1) less intLiteral(3)) and (stringLiteral("kotlin") like "kot%")
+        }
+        assertThat(comparison.toString()).isEqualTo("(1 < 3) AND ('kotlin' LIKE 'kot%')")
+
+        val inList = with(SqlExpressionBuilder) { intLiteral(2) inList listOf(3, 1, 2) }
+        assertThat(inList.toString()).isEqualTo("2 IN (3, 1, 2)")
+
+        val arithmetic = with(SqlExpressionBuilder) { ((intLiteral(5) + 4) * intLiteral(2)) % 3 }
+        assertThat(arithmetic.toString()).isEqualTo("(((5 + 4) * 2) % 3)")
+
+        val caseExpression = with(SqlExpressionBuilder) {
+            case()
+                .When(intLiteral(1) eq 1, stringLiteral("one"))
+                .Else(stringLiteral("other"))
+        }
+        assertThat(caseExpression.toString()).isEqualTo("CASE WHEN 1 = 1 THEN 'one' ELSE 'other' END")
+
+        val coalesced = with(SqlExpressionBuilder) {
+            coalesce(stringLiteral("primary"), stringLiteral("fallback"), stringLiteral("last"))
+        }
+        assertThat(coalesced.toString()).isEqualTo("COALESCE('primary', 'fallback', 'last')")
+
+        val aliased = intLiteral(10).alias("ten")
+        assertThat(aliased.toString()).isEqualTo("10 ten")
+        assertThat(aliased.aliasOnlyExpression().toString()).isEqualTo("ten")
+
+        val window = with(SqlExpressionBuilder) {
+            rowNumber()
+                .over()
+                .partitionBy(stringLiteral("partition"))
+                .rows(WindowFrameBound.unboundedPreceding())
+        }
+        assertThat(window.toString()).isEqualTo("ROW_NUMBER() OVER(PARTITION BY 'partition' ROWS UNBOUNDED PRECEDING)")
+    }
+
+    @Test
+    fun versionParsingComparesSemanticComponents(): Unit {
+        val version: Version = Version.from("1.2.0-beta-4")
+
+        assertThat(version.toString()).isEqualTo("1.2.0")
+        assertThat(version.covers("1.1.9")).isTrue()
+        assertThat(version.covers("1.2.0")).isTrue()
+        assertThat(version.covers("1.2.1")).isFalse()
+        assertThat(Version.from("2").covers(1, 9, 9)).isTrue()
+    }
+
+    @Test
+    @OptIn(ExperimentalKeywordApi::class)
+    fun databaseConfigBuilderAppliesDefaultsCustomizationsAndValidation(): Unit {
+        val defaultConfig: DatabaseConfig = DatabaseConfig()
+
+        assertThat(defaultConfig.useNestedTransactions).isFalse()
+        assertThat(defaultConfig.defaultFetchSize).isNull()
+        assertThat(defaultConfig.defaultIsolationLevel).isEqualTo(-1)
+        assertThat(defaultConfig.defaultMaxAttempts).isEqualTo(3)
+        assertThat(defaultConfig.defaultMinRetryDelay).isZero()
+        assertThat(defaultConfig.defaultMaxRetryDelay).isZero()
+        assertThat(defaultConfig.defaultReadOnly).isFalse()
+        assertThat(defaultConfig.warnLongQueriesDuration).isNull()
+        assertThat(defaultConfig.maxEntitiesToStoreInCachePerEntity).isEqualTo(Int.MAX_VALUE)
+        assertThat(defaultConfig.keepLoadedReferencesOutOfTransaction).isFalse()
+        assertThat(defaultConfig.explicitDialect).isNull()
+        assertThat(defaultConfig.defaultSchema).isNull()
+        assertThat(defaultConfig.logTooMuchResultSetsThreshold).isZero()
+        assertThat(defaultConfig.preserveKeywordCasing).isTrue()
+
+        val schema = Schema("analytics")
+        val customConfig: DatabaseConfig = DatabaseConfig {
+            useNestedTransactions = true
+            defaultFetchSize = 64
+            defaultIsolationLevel = 8
+            defaultMaxAttempts = 2
+            defaultMinRetryDelay = 10L
+            defaultMaxRetryDelay = 20L
+            defaultReadOnly = true
+            warnLongQueriesDuration = 250L
+            maxEntitiesToStoreInCachePerEntity = 5
+            keepLoadedReferencesOutOfTransaction = true
+            defaultSchema = schema
+            logTooMuchResultSetsThreshold = 3
+            preserveKeywordCasing = false
+        }
+
+        assertThat(customConfig.useNestedTransactions).isTrue()
+        assertThat(customConfig.defaultFetchSize).isEqualTo(64)
+        assertThat(customConfig.defaultIsolationLevel).isEqualTo(8)
+        assertThat(customConfig.defaultMaxAttempts).isEqualTo(2)
+        assertThat(customConfig.defaultMinRetryDelay).isEqualTo(10L)
+        assertThat(customConfig.defaultMaxRetryDelay).isEqualTo(20L)
+        assertThat(customConfig.defaultReadOnly).isTrue()
+        assertThat(customConfig.warnLongQueriesDuration).isEqualTo(250L)
+        assertThat(customConfig.maxEntitiesToStoreInCachePerEntity).isEqualTo(5)
+        assertThat(customConfig.keepLoadedReferencesOutOfTransaction).isTrue()
+        assertThat(customConfig.defaultSchema).isEqualTo(schema)
+        assertThat(customConfig.logTooMuchResultSetsThreshold).isEqualTo(3)
+        assertThat(customConfig.preserveKeywordCasing).isFalse()
+
+        assertThatThrownBy { DatabaseConfig { defaultMaxAttempts = 0 } }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("defaultMaxAttempts")
+    }
+}
+
+private object Users : Table("app.users") {
+    val id = integer("id").autoIncrement()
+    val email = varchar("email", 128).uniqueIndex("ux_users_email")
+    val displayName = varchar("display_name", 64).nullable()
+    val status = varchar("status", 32).default("active")
+    val score = integer("score").clientDefault { 7 }
+    val active = bool("active").default(true)
+    val generatedToken = varchar("generated_token", 32).clientDefault { "generated" }.databaseGenerated()
+
+    override val primaryKey = PrimaryKey(id, name = "pk_users")
+}
+
+private object Groups : Table("groups") {
+    val id = integer("id")
+    val name = varchar("name", 80)
+
+    override val primaryKey = PrimaryKey(id)
+}
+
+private object Memberships : Table("memberships") {
+    val user = reference("user_id", Users.id)
+    val group = reference("group_id", Groups.id)
+    val role = varchar("role", 32).default("member")
+
+    override val primaryKey = PrimaryKey(user, group, name = "pk_memberships")
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/user-code-filter.json
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.2.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.jetbrains.exposed.v1.**"
+    },
+    {
+      "includeClasses" : "org_jetbrains_exposed.exposed_core.**"
+    }
+  ]
+}

